src/OrchardCore.Modules/Orchard.Layers/Controllers/AdminController.cs

```diff
@@ -261,8 +261,31 @@ INotifier notifier
 
 			_session.Save(contentItem);
 
-			// Clear the cache
-			_signal.SignalToken(LayerMetadataHandler.LayerChangeToken);
+            // In case the moved contentItem is the draft for a published contentItem we update it's position too.
+            // We do that because we want the position of published and draft version to be the same.
+            if (contentItem.IsPublished() == false)
+            {
+                var publishedContentItem = await _contentManager.GetAsync(contentItemId, VersionOptions.Published);
+                if (publishedContentItem != null)
+                {
+                    layerMetadata = contentItem.As<LayerMetadata>();
+
+                    if (layerMetadata == null)
+                    {
+                        return StatusCode(403);
```

Comment:
Just do nothing in this case: just wrap the position/zone update in a layerMetadata != null